File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s

        i, d = 0, 1
        rows = [[] for _ in range(numRows)]

        for c in s:
            rows[i].append(c)
            if i == 0:
                d = 1
            elif i == numRows - 1:
                d = -1
            i += d
        
        res = ""
        for i in range(numRows):
            res +="".join(rows[i])

        return res

File: 0006-zigzag-conversion/0006-zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1) {
            return s;
        }

        int i = 0, d = 1;
        vector<string> rows(numRows);

        for (char c : s) {
            rows[i] += c;
            if (i == 0) {
                d = 1;
            } else if (i == numRows - 1) {
                d = -1;
            }
            i += d;
        }

        string res = "";
        for (const string& row : rows) {
            res += row;
        }

        return res;
    }
};
